### ISSUE TITLE
Variables beginning with `$` are not supported

### ISSUE THREAD

**Issue by quoll** [yetanalytics/flint#42]
`format-query` on a query using `$` vars are reported as a syntax error:

```clojure
(f/format-query '{:prefixes {:ex "<http://ex.com/>"}
                  :select [$e]
                  :where [[$e :ex/ref ?a]]}
                :pretty? true)
```
Throws with: `Syntax errors exist in the SELECT and WHERE clauses!`

```clojure
(f/format-query '{:prefixes {:ex "<http://ex.com/>"}
                  :select [?e]
                  :where [[?e :ex/ref $a]]}
                :pretty? true)
```
Throws with: `Syntax errors exist in the WHERE clause!`
---
This is syntax on the Clojure side, which is defined by Flint. So it is not strictly a bug. However, it is unexpected.

Since elements like variable names get copied directly into SPARQL, then I would expect `$a` to be accepted and show up in the SPARQL the same way.

**Comment by milt** [comment]
Thanks Paula, I had no idea that the variable sigil could be `$` in SPARQL. I also didn't know that:

> In a query, $abc and ?abc identify the same variable.

[4.1.3 here](https://www.w3.org/TR/sparql11-query/)

Given that I think in addition to adding support for the `$` sigil we'll have to rework things a bit to support comparing variable names canonically w/o the sigil.

**Comment by milt** [comment]
Added support in v0.3.1
